solutions/src/main/java/ru/sberbank/school/task08/KryoSerializableManager.java

```diff
@@ -0,0 +1,84 @@
+package ru.sberbank.school.task08;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import lombok.NonNull;
+import ru.sberbank.school.task08.kryo.MapStateSerializer;
+import ru.sberbank.school.task08.state.GameObject;
+import ru.sberbank.school.task08.state.InstantiatableEntity;
+import ru.sberbank.school.task08.state.MapState;
+import ru.sberbank.school.task08.state.Savable;
+import ru.sberbank.school.util.Solution;
+
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Solution(8)
+public class KryoSerializableManager extends SaveGameManager {
+
+    private Kryo kryo;
+
+    /**
+     * Класс-наследник должен иметь конструктор эквивалентный этому.
+     *
+     * @param filesDirectoryPath Путь до директории, в которой хранятся файлы сохранений
+     */
+    public KryoSerializableManager(@NonNull String filesDirectoryPath) {
+        super(filesDirectoryPath);
+    }
+
+    @Override
+    public void initialize() {
+        this.kryo = new Kryo();
+        kryo.register(MapState.class, new MapStateSerializer());
+        kryo.register(GameObject.class);
+        kryo.register(ArrayList.class);
+        kryo.register(InstantiatableEntity.Status.class);
+        kryo.register(InstantiatableEntity.Type.class);
+    }
+
+
+    @Override
+    public void saveGame(String filename, Savable gameState) throws SaveGameException {
+
+        try (OutputStream outputStream = new FileOutputStream(filesDirectory + filename);
+               Output output = new Output(outputStream)) {
+           kryo.writeObject(output, gameState);
```

Comment:
&#39;try&#39; child has incorrect indentation level 11, expected level should be 12.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a263815377>*